### DISCUSSION TITLE
flux multi-tenancy minimum supported version

### DISCUSSION THREAD

**Discussion by dariuslwh95** [fluxcd/flux2#3651] [Q&A] [answered]
hi I have been exploring the use case of soft multi-tenancy and multi-tenancy-lockdown from https://github.com/fluxcd/flux2-multi-tenancy and https://fluxcd.io/flux/installation/#multi-tenancy-lockdown

**is 0.26 the minimum supported version for flux-multi-tenancy?**

I have tried using the serviceaccountname with various roles bound to said specified account, however no resources have been patched to the cluster despite reconciling sources and experimenting with role types e.g. cluster-admin, default

only upon removal of the spec "serviceAccountName" do resources start to be applied, the spec "targetNamespace" does not seem to achieve the desired effect as well. 

I am currently using flux v0.23. 
apiVersion: kustomize.toolkit.fluxcd.io/v1beta2
kind: Kustomization
metadata:
  name: tenant-1
  namespace: tenant-1
spec:
  interval: 15m0s
  path: ./base
  prune: true
  targetNamespace: tenant-1
  serviceAccountName: t1-flux-serviceaccount
  sourceRef:
    kind: GitRepository
    name: tenant-1

**Comment by stefanprodan** [comment, marked as answer]
Please try to keep Flux up to date ([see here how to automate upgrade](https://github.com/fluxcd/flux2/tree/main/action#automate-flux-updates)), you should be using 0.40, 0.23 is affected by serious CVEs.

To answer your question: Yes 0.26 introduces multi-tenancy lockdown.

**Reply by dariuslwh95** [reply]
thanks for the reply Stefan, will upgrade, to be more specific is the impersonation of the serviceaccount role supported in flux 0.23? i’ve not been able to get resources deployed even with a serviceaccount bound with a cluster-admin role

**Reply by stefanprodan** [reply]
I don't remember, feel free to look over the changelogs. FYI using cluster-admin in a RoleBinding doesn't grant permissions outside of the namespace where the RoleBinding is. Also Kubernetes no longer provisions tokens to SAs, so whatever we did years ago in Flux it no longer works, please use latest 0.40.
